src/megaapi_impl.cpp

```diff
@@ -12920,27 +13017,114 @@ bool MegaApiImpl::sync_syncable(Sync *sync, const char *name, string *localpath)
     return result;
 }
 
-void MegaApiImpl::sync_auto_resumed(const string& localPath, const handle remoteNode, const long long localFp, const std::vector<std::string>& regExp)
+void MegaApiImpl::sync_removed(int tag)
 {
-    const int nextTag = client->nextreqtag();
+    eraseSync(tag);
+}
 
-    MegaSyncPrivate *sync = new MegaSyncPrivate(localPath.c_str(), remoteNode, -nextTag);
-    sync->setLocalFingerprint(localFp);
+void MegaApiImpl::sync_auto_resume_result(const SyncConfig &config, const syncstate_t &state, const syncerror_t &syncError)
```

Comment:
I will make a judgment call here and decide not to change this :-P, so that the prototype reflects the constness contract between the caller. This function is called once per sync in the entire execution. Performance in this is not importan. In any case, thanks for pointing that out. Will surely come handy elsewhere ;-)

Reply:
You can keep the `const`, but avoid the reference, in order to reflect the constness contract between the caller.
Nevertheless, I agree it's not going to make any difference in performance for this case, but a good practice. Feel free to amend, if you like it.